### DISCUSSION TITLE
OpenWB in SHM visualisieren

### DISCUSSION THREAD

**Discussion by Finders83** [camueller/SmartApplianceEnabler#287] [Installation und Betrieb]
Servus miteinander, ich versuche gerade vergeblich meine OpenWB in den SHM zu bekommen.

Der SmartApplianceEnabler läuft auf einem Raspberry Pi4, dieser wurde per automatischem Rollout installiert. Im Ergebnis wird mir der Dienst "smartapplianceenabler.service" als running angezeigt:
![Status_Pi](https://user-images.githubusercontent.com/108220972/175812666-1b74da48-5be2-4fb3-8201-a4d10e4e0107.JPG)

Habe nunmehr die OpenWB wie im [openWB Forum](https://openwb.de/forum/viewtopic.php?p=16770#p16770) beschrieben im SAE angelegt. 
[
![SAE_neu](https://user-images.githubusercontent.com/108220972/208244862-0f01dc01-6d19-47b7-badc-0039d6a0296a.JPG)
](url)
![SAE_OpenWB_Zahler](https://user-images.githubusercontent.com/108220972/175812750-cb1e25c2-f895-49ca-97b2-194d95c9c79d.JPG)
Im Ergebnis wird mir die OpenWB nicht im Menü Status angezeigt "Keine steuerbaren Geräte vorhanden.". Lediglich in der Baumstruktur rechts finde ich die angelegte Box. Auch im SunnyPortal finde ich die OpenWB nicht.
![SAE](https://user-images.githubusercontent.com/108220972/175812665-2f621a14-a11e-42b6-a2d1-ef742042c09d.JPG)
Alle Geräte hängen im gleichen Netz und auch am gleichen Router. IP seitig kann ich vom Raspberry alle Geräte erreichen.

Könnt ihr mich bei der Lösungsfindung unterstützen?

Vielen Dank schon mal.

**Comment by camueller** [comment]
Deinen Screnshots zufolge hast Du ein Gerät konfiguriert, das "OpenWallbox" heisst und einen Zähler dafür. Wenn Du die Wallbox mit dem SAE nur visualisieren willst, darfst Du nicht den Typ "Elektoauto-Ladegerät" verwenden, weil hier die Konfiguration einer Wallbox erwartet wird (die Du ja nicht im SAE machen willst). Stattdessen kannst Du als Typ "Sonstige" angeben. Einen Schalter solltest Du nicht konfigurieren - dadurch wird der Schaltzustand aus den Zählerabfragen ermittelt.

**Comment by Finders83** [comment]
Danke für dein Feedback. Auch mit dem Typ "Sonstiges" will der SHM nicht mit dem SAE kommunizieren. Kann ich am Pi die "Veröffentlichung" der WallBox über SAE irgendwie prüfen?

**Comment by camueller** [comment]
https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/QA_DE.md

**Reply by Finders83** [reply]
Was mir noch einfällt. Ich habe nach der Installation des Raspberry von WiFi auf Kabel gewechselt? Hängt der SAE-Dienst jetzt evtl. auf der WiFi Schnittstelle?

**Comment by camueller** [comment]
Wenn Du den SAE nicht fest auf eine Schnittstelle gebunden hast, sollte das egal sein.

**Reply by Finders83** [reply]
Habe die Schnitte nunmehr manuell konfiguriert und nunmehr ist die OpenWB in SunnyPortal gewandert:
![SAE_OpenWB](https://user-images.githubusercontent.com/108220972/176392918-92c5da78-fa67-4373-992c-a080809a40db.JPG)

Danke für deine Unterstützung

**Reply by JensHoeppner** [reply]
Magst du deine Konfiguration mal zeigen, würde das nämlich auch umsetzen wollen.
Danke dir

**Reply by Finders83** [reply]
Was genau möchtest du wissen?

**Reply by JensHoeppner** [reply]
Wie hast du dein Gerät im SAE konfiguriert?
Der SAE läuft bei mir, ich komme auf die Konfigurationsseite. Die OpenWB läuft ebenfalls. Die Steuerung der WB wird von FHEM aus gemacht und läuft ebenfalls zufriedenstellend. Es geht mir nur um die Visualisierung der Leistungsdaten der Wallbox im SMA-Portal.
Soweit ich gelesen habe, muss die Wallbox dann als "sonstiges Gerät" angelegt werden, da keine Steuerung passieren soll.
Ich habe also ein neues Gerät angelegt, bei dem ich nun den Zähler konfigurieren muss. Und da hänge ich nun: 
Ich würde tippen, dass ein http-Aufruf alle x Sekunden im JSON Format nötig ist. Benutzername und Passwort meine WB ist natürlich bekannt, aber glaube ich für den API Aufruf nicht nötig. Zumindest bekomme ich im Browser ohne Anmeldung bei Aufruf von http://<ip-der-WB/openWB/web/api.php?get=all diverse Werte zurückgeliefert. 
Aber wie geht es dann weiter?  Parameter: Leistung? oder Zählerstand?, Methode: get, Daten: lla1LP1 ? Pfad für Extraktion: ?? Regex für Extraktion ??
Oder kann der API Call angepasst werden, sodass nur noch der gesuchte Wert zurückgegeben wird und daher nichts mehr zu extrahieren ist?

**Comment by Finders83** [comment]
Erstelle einfach im SAE ein neues Gerät mit den Daten wie in meinem Screenshot zu sehen. Eine eindeutige ID musst du dir selbst nach dem gleichen Schema ausdenken. Die Zählerkonfiguration kannst du auch aus meinem Screenshot abschreiben. Danach gehst du ins Sunny Portal und suchst dort ein neues Gerät. 
Hier noch das Ergebnis aus dem SMA Portal:
![tempsnip](https://user-images.githubusercontent.com/108220972/208312364-95aa27bd-b420-4a2e-8aa0-5ff9578f3c77.png)

Fertig...

**Reply by JensHoeppner** [reply]
Ok, die Daten der Zählerkonfiguration habe ich eingetragen. Im Sunny Portal wird das Gerät gefunden.
Werde es so mal testen.
Danke dir schon mal.
